src/libcore/num/f32.rs

```diff
@@ -242,4 +242,32 @@ impl Float for f32 {
         let value: f32 = consts::PI;
         self * (value / 180.0f32)
     }
+
+    /// Returns the maximum of the two numbers.
+    #[inline]
+    fn max(self, other: f32) -> f32 {
+        // IEEE754 says: maxNum(x, y) is the canonicalized number y if x < y, x if y < x, the
+        // canonicalized number if one operand is a number and the other a quiet NaN. Otherwise it
+        // is either x or y, canonicalized (this means results might differ among implementations).
+        // When either x or y is a signalingNaN, then the result is according to 6.2.
+        //
+        // Since we do not support sNaN in Rust yet, we do not need to handle them.
+        // FIXME(nagisa): due to https://bugs.llvm.org/show_bug.cgi?id=33303 we canonicalize by
+        // multiplying by 1.0. Should switch to the `canonicalize` when it works.
+        (if self < other || other.is_nan() { self } else { other }) * 1.0
```

Comment:
@bombless 😂 

I was commenting on [896a41a9](https://github.com/rust-lang/rust/pull/42430/commits/896a41a9765de0750f626c1727f674bdfe0f660b#diff-14f327fd5e3b91eb2e9af0690596d156R257), nagisa fixed it in the new commit b9ed1c6, GitHub somehow messed up and used the old comment referring to the latest change.